packages/http/src/client.ts

```diff
@@ -0,0 +1,131 @@
+        : request(url, { method: 'get' }, config),
+    put: (url: string, body: unknown, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      input && 'headers' in input
+        ? request(url, { method: 'put', ...input, body }, config)
+        : request(url, { method: 'put', body }, config),
+    post: (url: string, body: unknown, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      input && 'headers' in input
+        ? request(url, { method: 'post', ...input, body }, config)
+        : request(url, { method: 'post', body }, config),
+    delete: (url: string, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      input && 'headers' in input
+        ? request(url, { method: 'delete', ...input }, config)
+        : request(url, { method: 'delete' }, config),
+    options: (url: string, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      input && 'headers' in input
+        ? request(url, { method: 'options', ...input }, config)
+        : request(url, { method: 'options' }, config),
+    head: (url: string, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      input && 'headers' in input
+        ? request(url, { method: 'head', ...input }, config)
+        : request(url, { method: 'head' }, config),
+    patch: (url: string, body: unknown, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      input && 'headers' in input
+        ? request(url, { method: 'patch', ...input, body }, config)
+        : request(url, { method: 'patch', body }, config),
+    trace: (url: string, input?: Pick<IHttpRequest, 'headers'> | IHttpConfig, config?: IHttpConfig) =>
+      input && 'headers' in input
+        ? request(url, { method: 'trace', ...input }, config)
+        : request(url, { method: 'trace' }, config),
+  };
+};
+
+type PrismOutput = {
+  status: IHttpResponse['statusCode'];
+  statusText: string;
+  headers: IHttpResponse['headers'];
+  data: IHttpResponse['body'];
+  config: IHttpConfig;
+  request: IHttpRequest;
+  validations: IPrismOutput<IHttpRequest, IHttpResponse>['validations'];
+};
+
+type RequestFunction = (url: string, input: Omit<IHttpRequest, 'url'>, config?: IHttpConfig) => Promise<PrismOutput>;
+
+interface IRequestFunctionWithVerb {
+  (url: string, input: Pick<IHttpRequest, 'headers'>, config?: IHttpConfig): Promise<PrismOutput>;
+  (url: string, config?: IHttpConfig): Promise<PrismOutput>;
+}
+
+interface IRequestFunctionWithVerbWithBody {
+  (url: string, body: unknown, input: Pick<IHttpRequest, 'headers'>, config?: IHttpConfig): Promise<PrismOutput>;
+  (url: string, body: unknown, config?: IHttpConfig): Promise<PrismOutput>;
+}
+
+export type PrismHttp = {
+  request: RequestFunction;
+  get: IRequestFunctionWithVerb;
+  put: IRequestFunctionWithVerbWithBody;
+  post: IRequestFunctionWithVerbWithBody;
+  delete: IRequestFunctionWithVerb;
```

Comment:
Can we avoid verb and say method instead? It should never be verb, we should change any instance of verb we see.

Reply:
Done!

Reply:
Done